node/packages/aws-lambda-sdk/instrument/index.js

```diff
@@ -146,14 +153,15 @@ module.exports = (originalHandler, options = {}) => {
       } else {
         resolveResponseTags(outcomeResult);
         if (!serverlessSdk._settings.disableResponseMonitoring) {
-          writeResponse(outcomeResult, context);
+          serverlessSdk._deferredTelemetryRequests.push(writeResponse(outcomeResult, context));
         }
       }
 
       const endTime = process.hrtime.bigint();
       awsLambdaInvocationSpan.close({ endTime });
       awsLambdaSpan.close({ endTime });
-      writeTrace();
+      serverlessSdk._deferredTelemetryRequests.push(writeTrace());
+      await Promise.all(serverlessSdk._deferredTelemetryRequests);
```

Comment:
Does this mean that we are deferring sending all of the trace payloads until the root lambda span is closed? 🤔

Reply:
Ahh okay awesome 👍